src/app/core/services/navigation/nav-tree.service.ts

```diff
@@ -76,9 +76,12 @@ export abstract class NavTreeService<ContentT extends RoutedContentInfo> {
     shareReplay(1),
   );
 
+  private navigationRootElementId$ = new BehaviorSubject<string | undefined>(undefined);
   navigationNeighbors$: Observable<FetchState<NavigationNeighbors|undefined>> = this.state$.pipe(
+    repeatLatestWhen(this.navigationRootElementId$), // refresh when navigation root element changes
     mapStateData(navData => {
-      if (!navData.selectedElementId) return undefined;
+      const navigationRootElementId = this.navigationRootElementId$.value;
+      if (!navData.selectedElementId || navData.selectedElementId === navigationRootElementId) return undefined;
```

Comment:
That's not the spec " should only allow navigating under (**not including itself** and but including its descendants)"

Reply:
fixed